inception-recommendation/src/main/java/de/tudarmstadt/ukp/inception/recommendation/service/RecommendationServiceImpl.java

```diff
@@ -310,7 +322,23 @@ public void onRecommenderDelete(RecommenderDeletedEvent aEvent)
     private void triggerTrainingAndClassification(String aUser, Project aProject)
     {
         User user = userRepository.get(aUser);
-        scheduler.enqueueTask(user, aProject);
+
+        // Update the task count
+        Pair<User, Project> key = new ImmutablePair<>(user, aProject);
+        int count = trainingTaskCounter.getOrDefault(key, 0);
+
+        // If it is time for a selection task, we just start a selection task.
+        // The selection task then will start the training once its finished,
+        // i.e. we do not start it here.
+        if (count % TRAININGS_PER_SELECTION == 0) {
+            Task task = new SelectionTask(aProject, user);
+            schedulingService.enqueue(task);
+        } else {
+            Task task = new TrainingTask(user, aProject);
+            schedulingService.enqueue(task);
+        }
+
+        trainingTaskCounter.put(key, count + 1);
```

Comment:
I think incrementing the counter should be made an atomic synchronized operation.